backend/modules/simulation/router.py

```
@@ -50,35 +58,112 @@ async def linear_chat_stream(
+                async for chunk in service.stream_chat_message(
+                    current_user.id,
+                    request.user_progress_id,
+                    request.message,
+                    request.scene_id
+                ):
+                    yield chunk
+            except NotImplementedError:
+                logger.error(
+                    f"[SIMULATION_ROUTER] Streaming not implemented: session_id={session_id}, "
+                    f"user_id={current_user.id}, user_progress_id={request.user_progress_id}"
+                )
```

Ruff (0.14.10)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)